web/modules/custom/joinup_front_page/src/Plugin/Block/ExploreItemBlock.php

```diff
@@ -0,0 +1,203 @@
+   */
+  protected $entityManager;
+
+  /**
+   * The messenger.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * Constructs a new ExploreItemBlock instance.
+   *
+   * @param array $configuration
+   *   The plugin configuration, i.e. an array with configuration values keyed
+   *   by configuration option name. The special key 'context' may be used to
+   *   initialize the defined contexts by setting it to an array of context
+   *   values keyed by context names.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_manager
+   *   The entity manager.
+   * @param \Drupal\Core\Messenger\MessengerInterface $messenger
+   *   The messenger.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_manager, MessengerInterface $messenger) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityManager = $entity_manager;
+    $this->messenger = $messenger;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('messenger')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(): array {
+    $data = [
+      'solutions' => [
+        'name' => $this->t('Solutions'),
+        'data' => $this->getRdfEntity('solution'),
+        'url' => '/search?keys=&f[0]=type%3Asolution',
+      ],
+      'collections' => [
+        'name' => $this->t('Collections'),
+        'data' => $this->getRdfEntity('collection'),
+        'url' => '/search?keys=&f[0]=type%3Acollection',
+      ],
+      'news' => [
+        'name' => $this->t('News'),
+        'data' => $this->getContent('news'),
+        'url' => '/search?keys=&f[0]=type%3Anews',
+      ],
+      'events' => [
+        'name' => $this->t('Events'),
+        'data' => $this->getContent('events'),
+        'url' => '/search?keys=&f[0]=type%3Aevent',
+      ],
+    ];
+
+    return [
+      '#theme' => 'explore_block',
+      '#data' => $data,
+      '#cache' => [
+        'contexts' => $this->getCacheContexts(),
+        'tags' => $this->getCacheTags(),
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    return Cache::mergeContexts(parent::getCacheContexts(), ['route']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
+    $tags = [
+      'node:news',
+      'node:events',
+      'rdf_entity:solution',
+      'rdf_entity:collection',
+    ];
+
+    return Cache::mergeTags(parent::getCacheTags(), $tags);
+  }
```

Comment:
I understand the reasoning behind this but it seems dangerous for a block that is shown on the front page. Having this vary by the list cache tags of 4 of our most popular content types means that the front page will be invalidated every single time a change is made to any of these types. I can imagine that this will at times cause the cache to be invalidated multiple times per minute.

Let's keep this for now, but we need to discuss this with the business analyst team to find a good solution, for example a simple time based cache. I will make a comment in the Jira ticket to get some feedback.

If we end up switching to a time based cache this will become very interesting to test :sweat_smile: